homeassistant/components/acmeda/base.py

```diff
@@ -62,7 +62,7 @@ async def async_will_remove_from_hass(self) -> None:
     def notify_update(self) -> None:
         """Write updated device state information."""
         LOGGER.debug("Device update notification received: %s", self.name)
-        self.async_write_ha_state()
+        self.schedule_update_ha_state()
```

Comment:
It looks like the dep is async so there shouldn't be a thread safety concern here, why switch this?

Reply:
notify_update is called outside of the event loop causing this issue: https://github.com/home-assistant/core/issues/123509

Reply:
Why does an async library run the callbacks in the executor?

Reply:
It seems like `async_add_job` was based on a similar function from a very old version of Home Assistant https://github.com/home-assistant/core/blob/0.99.0/homeassistant/core.py#L289 and removed support for running code as an [async callback](https://developers.home-assistant.io/docs/asyncio_categorizing_functions#the-callback-function), and instead runs all callbacks in the executor.

Reply:
It looks like everything is expecting to be run in the event loop in the callbacks so you could remove the executor calls and replace it with `self.loop.call_soon(target, *args)`, but since it doesn't look like anything needs to delay until the next loop and add more latency, you could instead run the callback with an exception guard:

```python
            try:
                target(*args)
            except Exception:
                _LOGGER.exception("Error running job: %s", job)
```

Reply:
Sorry I'm not completely following, which call would be wrapped with the exception guard?

Reply:
I think we should leave this as-is and fix the library to use the above code instead of running its callback in the executor when calling `async_X_job` in the lib

Reply:
My understanding is that async_write_ha_state must only be called from a thread from the hass event loop. The library (aiopulse) itself uses asyncio and so has its own event loop that calls the callback. Unless I'm missing something I'm not exactly sure how the library could (or should) be modified to try and call async_X_job from a thread in the hass event loop other than by adding a job to the hass event loop?

Reply:
I'm could remove this change from this pull request to get the other fixes implemented but this will definitely leave the integration in a broken state.

Reply:
I've removed this change for the PR and will submit a new pull request to resolve this issue.

Reply:
If you can't change the library at this time, its ok to leave it in with a comment that there are multiple event loops running, and while its not ideal, it so to much to refactor now and will be fixed in the future